### ISSUE TITLE
Training gets stuck with a specific poses_bounds.npy 

### ISSUE THREAD

**Issue by kondela** [kwea123/nerf_pl#49]
I've been experimenting with your great implementation of NeRF during the last few days and I noticed that training gets stuck  at the very beginning when I use a specific `poses_bounds.npy` generated by LLFF repo on poses recovered by COLMAP. I tried to use the very same images and temporarily took `poses_bounds.npy` from a different scene and training did not get stuck.

To clarify, by stuck I mean that the console looks for following (waited over 30 minutes) until I kill the process. 
```
INFO:lightning:GPU available: True, used: True
INFO:lightning:CUDA_VISIBLE_DEVICES: [0]
val image is /home/xyz/Projects/data/nerf/test_calibration/images/e075_a000.jpg
Validation sanity check: 0it [00:00, ?it/s]  
```
Also `nvidia-smi` reports that GPU is being fully utilized although training is stuck.

I use following parameters in `train.py`:

```
--dataset_name
llff
--root_dir
/home/xyz/Projects/data/nerf/test_calibration
--N_importance
64
--spheric
--use_disp
--num_epochs
30
--batch_size
2048
--optimizer
adam
--img_wh
256
256
--lr
5e-4
--lr_scheduler
cosine
--exp_name
test_calibration
--num_gpus
1
```

These are the `poses_bounds.npy` that cause PyTorch Lightning Trainer to get stuck.
[poses_bounds.npy.zip](https://github.com/kwea123/nerf_pl/files/5771250/poses_bounds.npy.zip)

I was able to use the very same dataset and `poses_bounds.npy` in a concurrent implementation [krrish94/nerf-pytorch](https://github.com/krrish94/nerf-pytorch) although I did not verify the rendered images.
I can privately share entire dataset (along with COLMAP .bin files) if necessary :)

**Comment by kondela** [comment]
I was able to pin point it and it hangs on accessing `denoms` tensor in the following line:
https://github.com/kwea123/nerf_pl/blob/20d1670b2d6d64e8f2bde65e263531f7af5779a5/models/rendering.py#L51

**Comment by kondela** [comment]
Issue was unrelated to this implementation - it was a missmatch of camera poses and camera parameters when generating `poses_bounds.npy` that led to a NaN values that created a deadlock in tensors passed to `torchsearchsorted`

**Comment by kwea123** [comment]
Hi, is the data same as you shared [here](https://github.com/facebookresearch/NSVF/issues/25)? Glad you solved it.

**Comment by kondela** [comment]
Hey, no problem. It was a different dataset. The problem was that my intrinsics were incorrect and the fact that COLMAP's GUI bundle adjustment normalizes scene, see for more information https://github.com/colmap/colmap/issues/1094.

By the way, do you plan to look into reimplementing NSVF into vanilla pytorch, i.e. with no fairseq framework dependence?

**Comment by kwea123** [comment]
I've read his implementation but haven't fully understood yet. It doesn't seem obvious to adapt to vanilla pytorch, I think his implementation is a good start point (although it's still mysterious for me because there is no code comment...). I have other priorities at the moment (extend nerf to handle dynamic object), so this will not be on my todo list. Do you plan to work on this?
